### ISSUE TITLE
[Feature Request]  - Add support for multiple calendars

### ISSUE THREAD

**Issue by albocc** [lfos/calcurse#68]
This software is great and CalDAV works nice so far, but what I am really missing, which is keeping me from using calcurse, is the lack of support for multiple calendars.

It would be amazing, if someone could add a feature to keep multiple calendars – ideally with CalDAV support.

**Comment by lfos** [comment]
You can already use (and synchronize) calcurse with multiple calendars, as see #14 and #59. The only thing that does not work is displaying multiple calendars within a running calcurse instance at the same time.

**Comment by albocc** [comment]
Thanks for your reply!

What I meant was that there is no support to view multiple calendars at once, as you already noted. I saw that I could create multiple instances of calcurse with different configuration directories, but that is not the same, unfortunately.

**Comment by lfos** [comment]
Ok, leaving this open... For the next release, our main priority is making major improvements to the synchronization script. So this might be a good candidate for 4.5.0...

**Comment by raspbeguy** [comment]
I think this is a very important feature because it will enable dav and vdir features to be more consistent.

**Comment by demonking** [comment]
Is this still planned ?

**Comment by lfos** [comment]
> Is this still planned ? @lfos

Yes.

**Comment by pprw** [comment]
I think the ability to display several calendar (with different colors) on the same calcurse instance is very important. Most people have at least a calendar for work and another for personal stuff.

**Comment by lfos** [comment]
> I think the ability to display several calendar (with different colors) on the same calcurse instance is very important. Most people have at least a calendar for work and another for personal stuff.

I totally agree. However, this is also nontrivial to implement. Patches are welcome!

**Comment by lfos** [comment]
As discussed in a related thread on the mailing list, a first step in this direction would be adding support for additional read-and-edit-only calendars. You can specify one main data file. Any new items will be added to this main calendar file. Other items can be edited.

In a second phase, we can add support for switching the main calendar in a running instance (or, alternatively, always ask for the calendar to use when adding items).

**Comment by pprw** [comment]
> > I think the ability to display several calendar (with different colors) on the same calcurse instance is very important. Most people have at least a calendar for work and another for personal stuff.
> 
> I totally agree. However, this is also nontrivial to implement. Patches are welcome!

Of course, I always easier to ask than to implement. I just discovered calcurse and it is wonderful so far! Keep up the good work.

**Comment by jbh** [comment]
I would like to add my +1 to this, as it's the only feature missing keeping me from using calcurse as my main calendar software. I would even be satisfied with running multiple instances. That even sounded better and more organized, because I would have a calcurse instance per calendar, meaning I could theme each one differently to make it more clear what calendar I was working in. I was going to do that and dedicate  a workspace to my calendars, but it seems one cannot run multiple instances at the same time, so that's a pain. Is this the case, or am I missing something? When I try to run another instance, I get an error.

Anyway, I don't mean to complain. I really appreciate the work that went into this. It's a fantastic piece of software that I wish I could use as my daily driver. I'll be studying the source and seeing if I can contribute to this feature somehow.

**Comment by lhca** [comment]
> ... but it seems one cannot run multiple instances at the same time, so that's a pain. Is this the case, or am I missing something?

You are. See the `-D` option in the man page.

**Comment by jbh** [comment]
Dang. I swear I read through the man page. I remember reading that, but it wasn't clear to me that would solve this issue. I'll give it a shot. Thank you!

**Comment by jbh** [comment]
For anyone wondering later, you have to also pass `datadir` to calcurse-caldav, like so:

```
calcurse-caldav --init=keep-remote --config=$HOME/.calcurse-work/caldav/config --syncdb=$HOME/.calcurse-work/caldav/sync.db --datadir=$HOME/.calcurse-work
```

**Comment by ovgeorge** [comment]
@jbh could you describe how your final workflow looks like and how it works?

**Comment by jbh** [comment]
@ovgeorge I haven't used it in a very long time, because I just kinda stopped using calendars altogether.

I think my [dotfiles repo](https://github.com/jbh/dotfiles#features) has references to my workflow. Take a look at dmenu_calcurse.

I'll try to take a look over them this weekend to see if I can remember anything in particular.

**Comment by jbh** [comment]
@ovgeorge Looks like I was smart enough to commit configuration examples:

[calcurse configs](https://github.com/jbh/dotfiles/tree/master/.config/calcurse)
